### PR TITLE
[PLAT-6433] Additional compiler warnings

### PR DESCRIPTION
**Pull request by nickdowell**
## Goal

Enable additional compiler warnings to detect potential issues in our code, and address any warnings raised.

## Changeset

Refactors `Bugsnag.xcconfig` to use Xcode's warning build settings instead of raw compiler flags where available.

Enables the following additional warnings and fixes warnings raised:

* `-Wcast-qual`
* `-Wcustom-atomic-properties`
* `-Wdirect-ivar-access`
* `-Wdocumentation-unknown-command`
* `-Wformat-nonliteral`
* `-Widiomatic-parentheses`
* `-Wimport-preprocessor-directive-pedantic`
* `-Wmissing-variable-declarations`
* `-Woverriding-method-mismatch`
* `-Wswitch-enum`
* `-Wunused-macros`

The xcconfig is no longer applied to the unit test targets in order to avoid excessive warnings there. Some unit test code would by design result in build warnings due to testing edge cases & invalid arguments.

## Testing

Tested by running unit tests locally, and unit & E2E tests on CI.

### CODE REVIEW

**Review by kattrali — APPROVED** [review]

File: Bugsnag/Payload/BugsnagStackframe.m [hunk 1]
```diff
@@ -228,19 +228,19 @@ - (NSDictionary *)toDictionary {
     dict[BSGKeyMachoUUID] = self.machoUuid;
 
     if (self.frameAddress != nil) {
-        NSString *frameAddr = [NSString stringWithFormat:BSGKeyFrameAddrFormat, [self.frameAddress unsignedLongValue]];
+        NSString *frameAddr = [NSString stringWithFormat:@"0x%lx", [self.frameAddress unsignedLongValue]];
```
[the remark this answers is not in the document]
Comment (nickdowell) [reply Bugsnag/Payload/BugsnagStackframe.m]: 👉 https://github.com/bugsnag/bugsnag-cocoa/commit/741c8d91eeb6de4a6b77e57b82bf2e48fa367a90
